src/Illuminate/Testing/TestResponse.php

```diff
@@ -165,7 +166,7 @@ public function assertStatus($status)
     {
         $actual = $this->getStatusCode();
 
-        $lastException = $actual === 500 && $status !== 500
+        $lastException = $actual !== $status && in_array($actual, [422, 500])
                     ? $this->exceptions->last()
```

Comment:
This will always return `null` in the case of `ValidationException` since it's part of the base Handler internal don't report array:

https://github.com/laravel/framework/blob/63fd6adf289c1a24eb22075418c6abc334381d01/src/Illuminate/Foundation/Exceptions/Handler.php#L87-L98